Doc/library/importlib.rst

```diff
@@ -416,12 +416,16 @@ ABC hierarchy::
         Loaders that have a file-like storage back-end
         that allows storing arbitrary data
         can implement this abstract method to give direct access
-        to the data stored. :exc:`OSError` is to be raised if the *path* cannot
-        be found. The *path* is expected to be constructed using a module's
-        :attr:`__file__` attribute or an item from a package's :attr:`__path__`.
+        to the data stored.
+        The *path* is expected to be constructed using a module's
+        :attr:`__file__` attribute or an item from a package's
+        :attr:`__path__` attribute.
+
+        If the *path* cannot be handled, this raises an :exc:`OSError`
+        or a :exc:`ValueError` depending on the reason.
```

Comment:
```suggestion
        If the *path* cannot be handled, either :exc:`OSError` or :exc:`ValueError`
        is to be raised. In most cases, these will be raised by the underlying
        operating system interfaces rather than directly (e.g. :exc:`OSError`
        would be raised when attempting to access a valid but nonexistent
        filesystem path, while attemtping to access a path containing a NULL
        byte would raise :exc:`ValueError`).
```